nuleaf-source/controllers/posts.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * Posts Controller
+ * 
+ * This controller handles requests for creating, updating, deleting, and searching 
+ * posts. All responses are in JSON format.
+ */
+
+/* jslint node: true */
+'use strict';
+
+var PostsDAO = require('..database/posts');
+
+
+/**
+ * Search for posts in the database. A successful response will always
+ * be an array even if no steminars were found matching the filters.
+ *
+ * Parameters list:
+ *     title         : Filter for posts with this title.
+ *     content       : Filter for posts with this content.
+ *     author        : Filter for posts with this author.
+ *     date_created  : Filter for posts created on this date.
+ *     date_published: Filter for posts published on this date.
+ *     date_modified : Filter for posts modified on this date.
+ *     sort          : Stores a number that determines if the results are shown in de/ascending order.
+ *     sortBy        : Stores the attribute above that the results are sorted by.
+ *     skip          : Return a certain number of results after a certain number of documents.
+ *     limit         : Used to specify the maximum number of results to be returned.
+ */
+exports.search = function(req, res) {
+  PostsDao.find({
+    title         : req.query.title,
+    content       : req.query.content,
+    author        : req.query.author,
+    date_created  : req.query.date_created,
```

Comment:
These should be separated into two each: `req.query.created_after` and `req.query.created_before`, `req.query.published_after`, etc...